dask_expr/io/io.py

```diff
@@ -181,6 +183,168 @@ def _tune_up(self, parent):
+    }
+    _absorb_projections = False
+
+    @functools.cached_property
+    def _name(self):
+        if self.label is None:
+            return (
+                funcname(self.func).lower()
+                + "-"
+                + _tokenize_deterministic(*self.operands)
+            )
+        else:
+            return self.label + "-" + _tokenize_deterministic(*self.operands)
+
+    @functools.cached_property
+    def _meta(self):
+        if self.operand("user_meta") is not no_default:
+            meta = self.operand("user_meta")
+        else:
+            vals = [v[0] for v in self.iterables]
+            meta = self.func(*vals, *self.args, **self.kwargs)
+        return make_meta(meta)
+
+    def _divisions(self):
+        if self.operand("user_divisions"):
+            return self.operand("user_divisions")
+        else:
+            npartitions = len(self.iterables[0])
+            return (None,) * (npartitions + 1)
+
+    @property
+    def apply_func(self):
+        if self.enforce_metadata:
+            return apply_and_enforce
+        return self.func
+
+    @functools.cached_property
+    def apply_kwargs(self):
+        kwargs = self.kwargs
+        if self.enforce_metadata:
+            kwargs = kwargs.copy()
+            kwargs.update(
+                {
+                    "_func": self.func,
+                    "_meta": self._meta,
+                }
+            )
+        return kwargs
+
+    def _filtered_task(self, index: int):
+        vals = [v[index] for v in self.iterables]
+        if self.apply_kwargs:
+            return (apply, self.apply_func, vals + self.args, self.apply_kwargs)
+        else:
+            return (self.func, *vals, *self.args)
+
+
+class FromMapProjectable(FromMap):
+    _parameters = [
+        "func",
+        "iterables",
+        "columns",
+        "args",
+        "kwargs",
+        "user_meta",
+        "enforce_metadata",
+        "user_divisions",
+        "label",
+        "_partitions",
+        "_series",
+    ]
+    _defaults = {
+        "user_meta": no_default,
+        "enforce_metadata": False,
+        "user_divisions": None,
+        "label": None,
+        "_partitions": None,
+        "_series": False,
+    }
+    _absorb_projections = True
+
+    @functools.cached_property
+    def columns_operand(self):
+        return _convert_to_list(self.operand("columns"))
+
+    @property
+    def columns(self):
+        if self.columns_operand is None:
+            return list(self.frame_meta.columns)
+        else:
+            return self.columns_operand
+
+    @functools.cached_property
+    def _series(self):
+        # Only need to convert to _series if func
+        # doesn't produce a Series already
+        return self.operand("_series") and self.frame_meta.ndim > 1
+
+    @functools.cached_property
+    def kwargs(self):
+        options = self.operand("kwargs")
+        if self.columns_operand:
+            options["columns"] = self.columns_operand.copy()
```

Comment:
Sorry I meant both here, ``options`` is also an input as far as I understand? If you insert columns then this will propagate to the user

Reply:
Similar to what you did in ``def apply_kwargs(self):`` above